test/core/pipeline/test_reconnect.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+
+from haystack import Pipeline
+from haystack.core.errors import PipelineConnectError
+from haystack.testing.sample_components import Double
+
+
+def test_connect_component_twice():
```

Comment:
For completeness we may want to test that the implicit syntax also works (say, `pipe.connect("c1", "c2"))`) but that's not necessary.